app/Http/Controllers/Auth/OAuthController.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use App\User;
+use Exception;
+use Validator;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Foundation\Application;
+
+class OAuthController extends Controller
+{
+    private $app;
+
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Log in a user and returns an accessToken.
+     */
+    public function login(Request $request)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Reduce the number of returns of this function 5, down to the maximum allowed 3. ([php:S1142](https://sonarcloud.io/organizations/monicahq/rules#rule_key=php%3AS1142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=monica&pullRequest=1174&issues=AWLbqaNJ-pl6AHs2SAyF&open=AWLbqaNJ-pl6AHs2SAyF)

[ISSUE_KEY]: AWLbqaNJ-pl6AHs2SAyF